### PR TITLE
Moar Datepicker tweaks

### PR DESCRIPTION
**Pull request by WunderBart**
rizzo datepicker tweaks

### CODE REVIEW

File: app/assets/javascripts/lib/components/datepicker.js [hunk 1]
```diff
@@ -21,74 +21,106 @@ define([ "jquery", "picker", "pickerDate", "pickerLegacy" ], function($) {
 
   function Datepicker(args) {
     this.config = $.extend({}, defaults, args);
+    this.target = $(this.config.target);
 
-    this.init();
+    this.target.length && this.init();
+    this.listen();
```
Comment (tkali) [inline app/assets/javascripts/lib/components/datepicker.js]: Should it be here regardless of `this.target.length && this.init();` being true?

File: app/assets/javascripts/lib/components/datepicker.js [hunk 2]
```diff
@@ -21,74 +21,106 @@ define([ "jquery", "picker", "pickerDate", "pickerLegacy" ], function($) {
...
   Datepicker.prototype.init = function() {
+    var config = this.config,
+        options = this._getOptions();
+
+    this.inDate = this.target.find(config.startSelector);
+    this.outDate = this.target.find(config.endSelector);
+    this.inLabel = $(config.startLabelSelector);
+    this.outLabel = $(config.endLabelSelector);
+    this.firstTime = !!this.inDate.val();
+    this.day = 86400000;
```
Comment (tkali) [inline app/assets/javascripts/lib/components/datepicker.js]: Magic number of seconds in 24h?

File: app/assets/javascripts/lib/components/datepicker.js (hunk 1 above)
Comment (WunderBart) [inline app/assets/javascripts/lib/components/datepicker.js]: Noice catch. `listen` should be moved inside `init`. Silly me.

File: app/assets/javascripts/lib/components/datepicker.js (hunk 2 above)
Comment (WunderBart) [inline app/assets/javascripts/lib/components/datepicker.js]: Probably (not my code :trollface:)

**Comment by tkali** [comment]
:+1:

**Comment by nidalaa** [comment]
Clicky clicked in community :shipit:

**Pull request merged**